packages/td-tools/src/td-parser.ts

```diff
@@ -222,20 +222,20 @@ export function serializeTD(thing: Thing): string {
     const copy = JSON.parse(JSON.stringify(thing));
 
     // clean-ups
-    if (!copy.security || copy.security.length === 0) {
+    if (copy.security == null || copy.security.length === 0) {
```

Comment:
I think one error in the CI is actually caused by this line:

```suggestion
    if (copy?.security.length === 0) {
```

Reply:
@JKRhb Can you explain why this should cause issues? Locally the tests are all "green".
I even think the change you are proposing has a different meaning.

The current code 
`    if (!copy.security || copy.security.length === 0) {`
enters the if branch IF either `copy.security` is `null` or `undefined ` but also IF the security array is of `length === 0`

Your code misses the first part, doesn't it ?

Reply:
Hmm, yeah, you are right, I was mistaken with my suggestion, I thought the logic got reversed, but it should still be right. What got me to make the suggestion is that there is currently this error reported in the CI:

```
  1) the server side of servient
       should be able to add a Thing based on WoT.ThingFragment:
     TypeError: Cannot read properties of undefined (reading 'length')
      at Object.serializeTD (/home/runner/work/node-wot/node-wot/packages/td-tools/src/td-parser.ts:232:21)
      at ExposedThing.getThingDescription (src/exposed-thing.ts:126:30)
      at WoTServerTest.<anonymous> (test/ServerTest.ts:109:22)
      at Generator.next (<anonymous>)
      at fulfilled (test/ServerTest.ts:11:58)

```

However, I somehow made the suggestion to the wrong line. I would resolve this thread and try again 😅